fix(docs-infra): several style fixes for angular.io

gkalpak opened the pull request:
Several style fixes for angular.io:

<details>
  <summary><b>fix(docs-infra): clear unneeded DOM nodes in `CodeExample/TabsComponent`</b> <i>(click to expand)</i></summary><br />

  Both `CodeExampleComponent` and `CodeTabsComponent` components receive some code via content projection, glab the projected content and pass it through to a `CodeComponent` instance for formatting and displaying.

  Previously, the projected content was kept in the DOM (hidden). This unnecessarily increased the number of DOM nodes.

  This commit fixes this by clearing the projected DOM nodes once their content has been captured.

</details>

---
<details>
  <summary><b>fix(docs-infra): fix vertical alignment of contributor info items</b> <i>(click to expand)</i></summary><br />

  This commit fixes the vertical alignment of info items (i.e. links to bio, Twitter, website) in the contributor list.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106926453-ca407f80-6719-11eb-9580-8614cc4f4907.png" alt="contributor alignment before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106926469-cd3b7000-6719-11eb-8ba8-c8ae01996c02.png" alt="contributor alignment after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): dim line-numbers in code examples</b> <i>(click to expand)</i></summary><br />

  This commit ensures that the line-numbers in code examples appear "dimmed" to make it clearer that they are not part of the actual code.

  Based on the CSS rules, it seems that the intention was to make it so, but the color was incorrectly set on the `<ol>` elements instead of targeting `li::marker`.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106937191-23aeab80-6726-11eb-9178-bdc067fa3d09.png" alt="line-numbers before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106937184-227d7e80-6726-11eb-81d1-f709537b6a56.png" alt="line-numbers after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): fix search results top padding when a notification is visible</b> <i>(click to expand)</i></summary><br />

  Previously, when a notification was visible at the top of the page, the search results pane had more top padding than necessary.

  This commit fixes it by removing the extra padding.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106938103-3aa1cd80-6727-11eb-96f5-f607bde314be.png" alt="search-results after" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106938100-3970a080-6727-11eb-8ae8-4e7a33af1daf.png" alt="search-results before" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): avoid page-breaks inside an image when printing</b> <i>(click to expand)</i></summary><br />

  Previously, when printing, a page-break could be inserted inside an image (contained in a `.lightbox` element).

  This commit ensures no page-breaks are inserted inside `.lightbox` elements (which contain images).

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106939709-48f0e900-6729-11eb-912c-e5cd0cb40897.png" alt="print-page-break before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/106939707-48585280-6729-11eb-8d5f-a12312f32056.png" alt="print-page-break after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): improve line-spacing between search results (esp. for wrapped lines)</b> <i>(click to expand)</i></summary><br />

  Previously, the line-spacing between different search result itemswas the same as that between wrapped lines of the same result. This made it diffucult to distinguish the different results.

  This commit fixes it by reducing the line-spacing between wrapped lines of the same result item and keeping a larger line-spacing between different results.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107229685-ca48c400-6a26-11eb-8161-0c0768b10a2d.png" alt="search-results before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107229691-ccab1e00-6a26-11eb-9594-d4c37d0d72b2.png" alt="search-results after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): fix sidenav top position on narrow screens</b> <i>(click to expand)</i></summary><br />

  On larger screens the top-bar has a height of 64px. On screens smaller than 600px, the top-bar has a height of 56px. As a result, the sidenav should have a top position of 56px on screens smaller than 600px and 64px on other screens.

  Previously, the style setting the top position to 56px was tied to the presense of the `.collapsed` class, which depends on whether the sidenav is docked or not. The change from docked to collapsed sidenav, however, happens at 992px. As a result, e sidenav had an incorrect top possition (56px instead of 64px) on screens between 600px and 991px.

  This commit fixes this by ensuring the change of the top position for the sidenav happens at 600px.

</details>

---
<details>
  <summary><b>fix(docs-infra): improve line-spacing between TOC items (esp. for wrapped lines)</b> <i>cClick to expand)</i></summary><br />

  Previously, the line-spacing between different TOC items was almost the same as that between wrapped lines of the same item. This made it more diffucult to distinguish the different items.

  This commit fixes this by reducing the line-spacing between wrapped lines of the same item and keeping a larger line-spacing between different items.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107255272-ccb81780-6a40-11eb-9612-4e7b2058417d.png" alt="toc line-spacing before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107255280-cd50ae00-6a40-11eb-8f60-9ce8fba10e12.png" alt="toc line-spacing after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): fix placement of TOC hover/active markers (dots)</b> <i>(click to expand)</i></summary><br />

  This commit fixes the placement of the TOC hover/active markers (dots) so that they are horizontally in the middle of the marker "rail" and vertically in the middle of the first item line.

  This commit also refactor the `_toc.scss` to be more DRY (wrt ot styling the TOC markers/marker rail) and uses Sass variables and mixins to ensure that the styling is kept up-to-date wrt to future typography changes to TOC items.

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107270751-03e3f400-6a54-11eb-9d77-5b0a3cfb1c2b.png" alt="toc marker before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107270758-047c8a80-6a54-11eb-82c0-98b832c109a3.png" alt="toc marker after" height="250" />
  </p>

</details>

---
<details>
  <summary><b>fix(docs-infra): fix top padding of main content when a notification is visible</b> <i>(click to expand)</i></summary><br />

  This commit fixes the top padding of the main content on the homepage and other pages when a notification is visible at the top of the page.
  The problem was particularly obvious on the homepage (see images below).

  _**Before:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107394349-74e2e480-6b04-11eb-8853-71a153cdc459.png" alt="homepage top-right before" height="250" />
  </p>

  _**After:**_
  <p align="center">
    <img src="https://user-images.githubusercontent.com/8604205/107394336-71e7f400-6b04-11eb-9ef0-ff82f6f27875.png" alt="homepage top-right after" height="250" />
  </p>

</details>

mary-poppins commented:
You can preview 4ff29a8 at https://pr40802-4ff29a8.ngbuilds.io/.

mary-poppins commented:
You can preview 288483d at https://pr40802-288483d.ngbuilds.io/.

petebacondarwin commented:
Commit typo in **fix(docs-infra): fix placement of TOC hover/active markers (dots) **

This commit also refactor**s** the `_toc.scss` to be more DRY (wrt **to** styling

petebacondarwin commented:
Commit typos in: **fix(docs-infra): fix sidenav top position on narrow screens**

presen**c**e of the `.collapsed` class, which depends on whether the sidenav
...
happens at 992px. As a result, **th**e sidenav had an incorrect top pos~~s~~ition
(56px instead of 64px) on screens between 600px and 991px.

petebacondarwin commented:
Commit typo in **fix(docs-infra): improve line-spacing between search results (esp. for wrapped lines)**

Previously, the line-spacing between different search result items _ was
...
diff**i**cult to distinguish the different results.

petebacondarwin commented:
Typo in commit **fix(docs-infra): clear unneeded DOM nodes in `CodeExample/TabsComponent`**

some code via content projection, g**r**ab the projected content and pass it

petebacondarwin reviewed (approved):
Awesome improvements @gkalpak - I love love love the before and after screenshots.

gkalpak committed:
fix(docs-infra): dim line-numbers in code examples

This commit ensures that the line-numbers in code examples appear
"dimmed" to make it clearer that they are not part of the actual code.

Based on the CSS rules, it seems that the intention was to make it so,
but the color was incorrectly set on the `<ol>` elements instead of
targeting `li::marker`.

Before: ![line-numbers before][1]
After: ![line-numbers after][2]

[1]: https://user-images.githubusercontent.com/8604205/106937191-23aeab80-6726-11eb-9178-bdc067fa3d09.png
[2]: https://user-images.githubusercontent.com/8604205/106937184-227d7e80-6726-11eb-81d1-f709537b6a56.png

gkalpak committed:
fix(docs-infra): clear unneeded DOM nodes in `CodeExample/TabsComponent`

Both `CodeExampleComponent` and `CodeTabsComponent` components receive
some code via content projection, grab the projected content and pass it
through to a `CodeComponent` instance for formatting and displaying.

Previously, the projected content was kept in the DOM (hidden). This
unnecessarily increased the number of DOM nodes.

This commit fixes this by clearing the projected DOM nodes once their
content has been captured.

gkalpak committed:
fix(docs-infra): fix vertical alignment of contributor info items

This commit fixes the vertical alignment of info items (i.e. links to
bio, Twitter, website) in the contributor list.

Before: ![contributor alignment before][1]
After: ![contributor alignment after][2]

[1]: https://user-images.githubusercontent.com/8604205/106926453-ca407f80-6719-11eb-9580-8614cc4f4907.png
[2]: https://user-images.githubusercontent.com/8604205/106926469-cd3b7000-6719-11eb-8ba8-c8ae01996c02.png

gkalpak committed:
fix(docs-infra): improve line-spacing between search results (esp. for wrapped lines)

Previously, the line-spacing between different search result items was
the same as that between wrapped lines of the same result. This made it
difficult to distinguish the different results.

This commit fixes it by reducing the line-spacing between wrapped lines
of the same result item and keeping a larger line-spacing between
different results.

Before: ![search-results before][1]
After: ![search-results after][2]

[1]: https://user-images.githubusercontent.com/8604205/107229685-ca48c400-6a26-11eb-8161-0c0768b10a2d.png
[2]: https://user-images.githubusercontent.com/8604205/107229691-ccab1e00-6a26-11eb-9594-d4c37d0d72b2.png

gkalpak committed:
fix(docs-infra): avoid page-breaks inside an image when printing

Previously, when printing, a page-break could be inserted inside an
image (contained in a `.lightbox` element).

This commit ensures no page-breaks are inserted inside `.lightbox`
elements (which contain images).

Before: ![print-page-break before][1]
After: ![print-page-break after][2]

[1]: https://user-images.githubusercontent.com/8604205/106939709-48f0e900-6729-11eb-912c-e5cd0cb40897.png
[2]: https://user-images.githubusercontent.com/8604205/106939707-48585280-6729-11eb-8d5f-a12312f32056.png

gkalpak committed:
fix(docs-infra): fix sidenav top position on narrow screens

On larger screens the top-bar has a height of 64px. On screens smaller
than 600px, the top-bar has a height of 56px. As a result, the sidenav
should have a top position of 56px on screens smaller than 600px and
64px on other screens.

Previously, the style setting the top position to 56px was tied to the
presence of the `.collapsed` class, which depends on whether the sidenav
is docked or not. The change from docked to collapsed sidenav, however,
happens at 992px. As a result, the sidenav had an incorrect top position
(56px instead of 64px) on screens between 600px and 991px.

This commit fixes this by ensuring the change of the top position for
the sidenav happens at 600px.

gkalpak committed:
fix(docs-infra): fix search results top padding when a notification is visible

Previously, when a notification was visible at the top of the page, the
search results pane had more top padding than necessary.

This commit fixes it by removing the extra padding.

Before: ![search-results before][1]
After: ![search-results after][2]

[1]: https://user-images.githubusercontent.com/8604205/106938100-3970a080-6727-11eb-8ae8-4e7a33af1daf.png
[2]: https://user-images.githubusercontent.com/8604205/106938103-3aa1cd80-6727-11eb-96f5-f607bde314be.png

gkalpak committed:
fix(docs-infra): fix top padding of main content when a notification is visible

This commit fixes the top padding of the main content on the homepage
and other pages when a notification is visible at the top of the page.
The problem was particularly obvious on the homepage (see images below).

Before: ![homepage top-right before][1]
After: ![homepage top-right after][2]

[1]: https://user-images.githubusercontent.com/8604205/107394349-74e2e480-6b04-11eb-8853-71a153cdc459.png
[2]: https://user-images.githubusercontent.com/8604205/107394336-71e7f400-6b04-11eb-9ef0-ff82f6f27875.png

gkalpak committed:
fix(docs-infra): fix placement of TOC hover/active markers (dots)

This commit fixes the placement of the TOC hover/active markers (dots)
so that they are horizontally in the middle of the marker "rail" and
vertically in the middle of the item's first line.

This commit also refactors the `_toc.scss` to be more DRY (wrt styling
the TOC markers/marker rail) and uses Sass variables and mixins to
ensure that the styling is kept up-to-date wrt future typography changes
to TOC items.

Before: ![toc marker before][1]
After: ![toc marker after][2]

[1]: https://user-images.githubusercontent.com/8604205/107270751-03e3f400-6a54-11eb-9d77-5b0a3cfb1c2b.png
[2]: https://user-images.githubusercontent.com/8604205/107270758-047c8a80-6a54-11eb-82c0-98b832c109a3.png

gkalpak committed:
fix(docs-infra): improve line-spacing between TOC items (esp. for wrapped lines)

Previously, the line-spacing between different TOC items was almost the
same as that between wrapped lines of the same item. This made it more
diffucult to distinguish the different items.

This commit fixes this by reducing the line-spacing between wrapped
lines of the same item and keeping a larger line-spacing between
different items.

Before: ![toc line-spacing before][1]
After: ![toc line-spacing after][2]

[1]: https://user-images.githubusercontent.com/8604205/107255272-ccb81780-6a40-11eb-9612-4e7b2058417d.png
[2]: https://user-images.githubusercontent.com/8604205/107255280-cd50ae00-6a40-11eb-8f60-9ce8fba10e12.png

mary-poppins commented:
You can preview 0c3e9d1 at https://pr40802-0c3e9d1.ngbuilds.io/.
